components/Discussion/NotificationOptions.js

```diff
@@ -0,0 +1,251 @@
+import React, {PureComponent} from 'react'
+import PropTypes from 'prop-types'
+import { compose } from 'react-apollo'
+import { css } from 'glamor'
+import { CDN_FRONTEND_BASE_URL } from '../../lib/constants'
+import withT from '../../lib/withT'
+import {
+  isNotificationSupported,
+  getNotificationPermission
+} from '../../lib/utils/notification'
+import {
+  A,
+  Dropdown,
+  InlineSpinner
+} from '@project-r/styleguide'
+import Loader from '../Loader'
+import {
+  DISCUSSION_NOTIFICATION_OPTIONS,
+  withDiscussionPreferences,
+  withSetDiscussionPreferences,
+  webNotificationSubscription,
+  withUpdateNotificationSettings
+} from './enhancers'
+
+const styles = {
+  container: css({
+    marginTop: '20px',
+    position: 'relative'
+  }),
+  spinner: css({
+    position: 'absolute',
+    top: 10,
+    right: 5
+  })
+}
+
+class NotificationOptions extends PureComponent {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      mutating: false,
+      webNotificationsPermission: null
+    }
+  }
+
+  componentDidMount () {
+    this.subscribe()
+    this.maybeMute()
+  }
+
+  componentDidUpdate () {
+    this.subscribe()
+    this.maybeMute()
+  }
+
+  initNotificationsState () {
+    this.setState({
+      webNotificationsPermission: getNotificationPermission()
+    })
+  }
+
+  confirmPermission () {
+    if (!isNotificationSupported()) {
+      return
+    }
+    const { t, data, updateNotificationSettings } = this.props
+    const { discussionNotificationChannels } = data && data.me
+
+    window.Notification.requestPermission((status) => {
+      if (status !== 'granted') {
+        return
+      }
+      // Globally opt in to the WEB notification channel.
+      const channels = discussionNotificationChannels.indexOf('WEB') === -1
+        ? [...discussionNotificationChannels, 'WEB']
+        : discussionNotificationChannels
+      updateNotificationSettings({
+        discussionNotificationChannels: channels
+      })
+      /* eslint-disable no-new */
+      new window.Notification(t('components/Discussion/WelcomeNotification/title'), {
+        body: t('components/Discussion/WelcomeNotification/body'),
+        icon: `${CDN_FRONTEND_BASE_URL}/frontend/static/apple-touch-icon.png`
```

Comment:
`CDN_FRONTEND_BASE_URL` already includes `frontend`—rm `/frontend`. Same in the account setting one.